lib/form/html.js

```diff
@@ -90,6 +90,7 @@ const addRequired = (html, input, index) => {
 
     if (input.required === 'save' || inp.required === 'save' || input.required === 'publish' || inp.required === 'publish') {
       let level = '';
+      const required = input.required && inp.required;
```

Comment:
Updated `const required = inp.required || input.required`

Tried removing the conditional statement below which caused the test to fail so I am leaving it in for now. 
```      
if (inp.required !== undefined) {
        level += `required--${inp.required}`;
      }
      else {
        level += `required--${input.required}`;
      }
```